tfx/experimental/distributed_inference/graphdef_experiments/subgraph_partitioning/beam_pipeline_test.py

```diff
@@ -0,0 +1,125 @@
+
+
+class BeamPipelineTest(tf.test.TestCase):
+  """A test for the beam pipeline library."""
+
+  def test_validate_outputs(self):
+    """Compares the results from the original model and the beam pipeline."""
+    with tempfile.TemporaryDirectory() as temp_dir:
+      create_complex_graph.save_examples_as_graphdefs(temp_dir)
+
+      # Root graph is the graph to compute. Since it's unique, it's both the
+      # graph name extended and the graph name.
+      root_graph = 'main'
+      parent_graph_to_remote_graph_input_name_mapping = {
+          'main': {'remote_op_a': {'ids_a': 'ids1'},
+                   'remote_op_b': {'ids_b1': 'ids1',
+                                   'ids_b2': 'ids2'},
+                   'remote_op_a_1': {'ids_a': 'FloorMod_1'},
+                   'remote_op_b_1': {'ids_b1': 'FloorMod_1',
+                                     'ids_b2': 'FloorMod'}
+                  },
+          'remote_op_b': {'remote_op_a': {'ids_a': 'FloorMod'},
+                          'remote_op_a_1': {'ids_a': 'ids_b2'}
+                         }
+      }
+
+      # Create input PColl with this.
+      root_graph_inputs = [
+          {'main': {'import/ids1:0': 3, 'import/ids2:0': 3}},
+          {'main': {'import/ids1:0': 10, 'import/ids2:0': 10}}]
+
+      graph_name_to_filepath = {
+          'main': os.path.join(temp_dir, 'main_graph.pb'),
+          'remote_op_a': os.path.join(temp_dir, 'graph_a.pb'),
+          'remote_op_b': os.path.join(temp_dir, 'graph_b.pb')}
+      graph_name_to_outputs = {
+          'main': ['AddN_1'],
+          'remote_op_b': ['Add_1'],
+          'remote_op_a': ['embedding_lookup/Identity']}
+
+      original_model_outputs = _run_original_model(root_graph,
+                                                   root_graph_inputs,
+                                                   graph_name_to_filepath,
+                                                   graph_name_to_outputs)
+
+      graph_name_to_graph_def = graph_partition.get_graph_name_to_graph_def(
+          graph_name_to_filepath)
+      graph_name_to_specs = graph_partition.partition_all_graphs(
+          graph_name_to_graph_def, graph_name_to_outputs)
+
+      with test_pipeline.TestPipeline() as p:
+
+        inputs = p | 'LoadInputs' >> beam.Create(root_graph_inputs)
+        outputs = (inputs
+                   | 'RunModel' >> beam_pipeline.ExecuteOneGraph(
+                       graph_name_to_specs,
+                       parent_graph_to_remote_graph_input_name_mapping,
+                       root_graph)
+                   | 'ExtractOutputs' >> beam.Map(_extract_outputs,
+                                                  graph_name_to_outputs,
+                                                  root_graph))
+
+        # Problem: The output for the example graph is a scalar, equal_to
+        # doesn't work with more complex things like tensors.
+        util.assert_that(outputs, util.equal_to(original_model_outputs))
```

Comment:
Couldn't find a documentation for the optional argument equals_fn. Could you point me to one?

Reply:
https://github.com/apache/beam/blob/c739f92e7faeb6c7299b77b740b8900164819f3d/sdks/python/apache_beam/testing/util.py#L158

Reply:
Tried, it seems like with beam 2.23.0, util.equal_to() doesn't accept an equal_fn argument.

Reply:
This has been supported since Beam 2.20.0: https://github.com/apache/beam/blob/v2.20.0/sdks/python/apache_beam/testing/util.py#L163

Did you make a typo (`equals_fn` vs `equal_fn`) or could you have an older version of Beam installed?

Reply:
I had an older version of Beam instead.